File: misc/13_RomanToInteger.py
val = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}


class Solution:
    def romanToInt(self, s: str) -> int:
        total = 0
        for i in range(0, len(s) - 1):
            if val[s[i]] < val[s[i + 1]]:
                total -= val[s[i]]
            else:
                total += val[s[i]]
        return total + val[s[-1]]

File: misc/13_RomanToInteger.cpp
#include <string.h>

class Solution {
public:
    int romanToInt(std::string s) {
        int values[131];
        values['I'] = 1;
        values['V'] = 5;
        values['X'] = 10;
        values['L'] = 50;
        values['C'] = 100;
        values['D'] = 500;
        values['M'] = 1000;
        int total = values[s[s.length()-1]];
        for (int i = 0; i < s.length() - 1; i++) {
            if (values[s[i]] < values[s[i+1]]) {
                total -= values[s[i]];
            } else {
                total += values[s[i]];
            }
        }
        return total;
    }
};
